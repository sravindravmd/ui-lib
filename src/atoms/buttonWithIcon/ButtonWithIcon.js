import React from 'react'
import PropTypes from 'prop-types'
import { StyledIconButton } from './ButtonWithIcon.styles'
import { DownArrow, Add, Close, Edit } from '../../assets/images'
const Icons = { DownArrow, Add, Close, Edit }
export const ButtonWithIcon = ({
  text,
  icon,
  iconPosition,
  isRoundIcon,
  ...rest
}) => {
  const Icon = Icons[icon]

  if (isRoundIcon)
    return (
      <StyledIconButton {...rest} isRoundIcon={isRoundIcon}>
        {Icon && <Icon className='icon' />}
      </StyledIconButton>
    )
  return (
    <StyledIconButton {...rest}>
      {iconPosition && iconPosition === 'left' && Icon && (
        <Icon className='icon left-icon' />
      )}
      {text}
      {iconPosition && iconPosition === 'right' && Icon && (
        <Icon className='icon right-icon' />
      )}
    </StyledIconButton>
  )
}

ButtonWithIcon.defaultProps = {
  disabled: false,
  isPrimary: true,
  theme: 'none',
  emphasis: 'none',
  isRoundIcon: false,
}

ButtonWithIcon.propTypes = {
  text: PropTypes.string,
  onClick: PropTypes.func.isRequired,
  isPrimary: PropTypes.bool,
  isRoundIcon: PropTypes.bool,
  theme: PropTypes.oneOf(['none', 'dark', 'light']),
  emphasis: PropTypes.oneOf(['none', 'high', 'low']),
  icon: PropTypes.string,
  iconPosition: PropTypes.string,
}
