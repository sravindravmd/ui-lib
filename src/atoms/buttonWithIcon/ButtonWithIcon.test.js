import React from 'react'
import renderer from 'react-test-renderer'
import { ButtonWithIcon } from './ButtonWithIcon'

describe('Button', () => {
  it('renders primary Button + Icon correctly with default props', () => {
    const fn = jest.fn()
    const tree = renderer
      .create(
        <ButtonWithIcon
          text='Button Label'
          isPrimary={true}
          size='small'
          onClick={fn}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders primary Button correctly with size:small, theme: dark, isPrimary:true, iconPosition: right, icon: DownArrow, emphasis:high', () => {
    const fn = jest.fn()
    const tree = renderer
      .create(
        <ButtonWithIcon
          text='Button Label'
          isPrimary={true}
          size='small'
          onClick={fn}
          icon='DownArrow'
          iconPosition='right'
          emphasis='high'
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders primary Button correctly with size:small, theme: light, isPrimary:false, iconPosition: right, icon: Add, emphasis:low', () => {
    const fn = jest.fn()
    const tree = renderer
      .create(
        <ButtonWithIcon
          text='Button Label'
          isPrimary={false}
          size='small'
          theme='light'
          onClick={fn}
          icon='DownArrow'
          iconPosition='left'
          emphasis='low'
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Render Icon Button with isRoundIcon: true, isPrimary: true', async () => {
    const fn = jest.fn()
    const tree = renderer
      .create(
        <ButtonWithIcon
          isRoundIcon={true}
          isPrimary={true}
          icon='Add'
          onClick={fn}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Render Icon Button with isRoundIcon: true, isPrimary: false, emphasis: high, theme: light', async () => {
    const fn = jest.fn()
    const tree = renderer
      .create(
        <ButtonWithIcon
          isRoundIcon={true}
          isPrimary={true}
          emphasis='high'
          theme='light'
          icon='Add'
          onClick={fn}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('Render Icon Button with isRoundIcon: true, isPrimary: false, emphasis: low, theme: light', async () => {
    const fn = jest.fn()
    const tree = renderer
      .create(
        <ButtonWithIcon
          isRoundIcon={true}
          isPrimary={true}
          emphasis='low'
          theme='light'
          icon='Add'
          onClick={fn}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
