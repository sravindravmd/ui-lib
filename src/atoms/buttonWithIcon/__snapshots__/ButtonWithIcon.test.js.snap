// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Render Icon Button with isRoundIcon: true, isPrimary: false, emphasis: high, theme: light 1`] = `
<button
  className="sc-bdfBwQ kVfCiU"
  disabled={false}
  onClick={[MockFunction]}
>
  <svg
    className="icon"
    height="16"
    viewBox="0 0 16 16"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8.74999 2.00002C8.75 1.5858 8.41421 1.25001 8 1.25001C7.58579 1.25 7.25 1.58579 7.24999 2L8.74999 2.00002ZM7.24988 13.9999C7.24987 14.4141 7.58566 14.7499 7.99987 14.7499C8.41408 14.7499 8.74987 14.4142 8.74988 13.9999L7.24988 13.9999ZM7.24999 2L7.24988 13.9999L8.74988 13.9999L8.74999 2.00002L7.24999 2Z"
    />
    <path
      d="M14 8.74999C14.4142 8.75 14.75 8.41421 14.75 8C14.75 7.58579 14.4142 7.25 14 7.24999L14 8.74999ZM2.00007 7.24988C1.58586 7.24987 1.25007 7.58566 1.25006 7.99987C1.25006 8.41408 1.58584 8.74987 2.00005 8.74988L2.00007 7.24988ZM14 7.24999L2.00007 7.24988L2.00005 8.74988L14 8.74999L14 7.24999Z"
    />
  </svg>
</button>
`;

exports[`Button Render Icon Button with isRoundIcon: true, isPrimary: false, emphasis: low, theme: light 1`] = `
<button
  className="sc-bdfBwQ kVfCiU"
  disabled={false}
  onClick={[MockFunction]}
>
  <svg
    className="icon"
    height="16"
    viewBox="0 0 16 16"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8.74999 2.00002C8.75 1.5858 8.41421 1.25001 8 1.25001C7.58579 1.25 7.25 1.58579 7.24999 2L8.74999 2.00002ZM7.24988 13.9999C7.24987 14.4141 7.58566 14.7499 7.99987 14.7499C8.41408 14.7499 8.74987 14.4142 8.74988 13.9999L7.24988 13.9999ZM7.24999 2L7.24988 13.9999L8.74988 13.9999L8.74999 2.00002L7.24999 2Z"
    />
    <path
      d="M14 8.74999C14.4142 8.75 14.75 8.41421 14.75 8C14.75 7.58579 14.4142 7.25 14 7.24999L14 8.74999ZM2.00007 7.24988C1.58586 7.24987 1.25007 7.58566 1.25006 7.99987C1.25006 8.41408 1.58584 8.74987 2.00005 8.74988L2.00007 7.24988ZM14 7.24999L2.00007 7.24988L2.00005 8.74988L14 8.74999L14 7.24999Z"
    />
  </svg>
</button>
`;

exports[`Button Render Icon Button with isRoundIcon: true, isPrimary: true 1`] = `
<button
  className="sc-bdfBwQ inFlso"
  disabled={false}
  onClick={[MockFunction]}
>
  <svg
    className="icon"
    height="16"
    viewBox="0 0 16 16"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8.74999 2.00002C8.75 1.5858 8.41421 1.25001 8 1.25001C7.58579 1.25 7.25 1.58579 7.24999 2L8.74999 2.00002ZM7.24988 13.9999C7.24987 14.4141 7.58566 14.7499 7.99987 14.7499C8.41408 14.7499 8.74987 14.4142 8.74988 13.9999L7.24988 13.9999ZM7.24999 2L7.24988 13.9999L8.74988 13.9999L8.74999 2.00002L7.24999 2Z"
    />
    <path
      d="M14 8.74999C14.4142 8.75 14.75 8.41421 14.75 8C14.75 7.58579 14.4142 7.25 14 7.24999L14 8.74999ZM2.00007 7.24988C1.58586 7.24987 1.25007 7.58566 1.25006 7.99987C1.25006 8.41408 1.58584 8.74987 2.00005 8.74988L2.00007 7.24988ZM14 7.24999L2.00007 7.24988L2.00005 8.74988L14 8.74999L14 7.24999Z"
    />
  </svg>
</button>
`;

exports[`Button renders primary Button + Icon correctly with default props 1`] = `
<button
  className="sc-bdfBwQ cJqlRs"
  disabled={false}
  onClick={[MockFunction]}
  size="small"
>
  Button Label
</button>
`;

exports[`Button renders primary Button correctly with size:small, theme: dark, isPrimary:true, iconPosition: right, icon: DownArrow, emphasis:high 1`] = `
<button
  className="sc-bdfBwQ cJqlRs"
  disabled={false}
  onClick={[MockFunction]}
  size="small"
>
  Button Label
  <svg
    className="icon right-icon"
    height="12"
    viewBox="0 0 12 12"
    width="12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2.70711 5.70711C2.07714 5.07714 2.52331 4 3.41421 4L8.58579 4C9.47669 4 9.92286 5.07714 9.29289 5.70711L6.70711 8.29289C6.31658 8.68342 5.68342 8.68342 5.29289 8.29289L2.70711 5.70711Z"
    />
  </svg>
</button>
`;

exports[`Button renders primary Button correctly with size:small, theme: light, isPrimary:false, iconPosition: right, icon: Add, emphasis:low 1`] = `
<button
  className="sc-bdfBwQ fHQcaw"
  disabled={false}
  onClick={[MockFunction]}
  size="small"
>
  <svg
    className="icon left-icon"
    height="12"
    viewBox="0 0 12 12"
    width="12"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2.70711 5.70711C2.07714 5.07714 2.52331 4 3.41421 4L8.58579 4C9.47669 4 9.92286 5.07714 9.29289 5.70711L6.70711 8.29289C6.31658 8.68342 5.68342 8.68342 5.29289 8.29289L2.70711 5.70711Z"
    />
  </svg>
  Button Label
</button>
`;
